### ISSUE TITLE
JS code generation does not work with single-field structures

### ISSUE THREAD

**Issue by eggrobin** [rust-diplomat/diplomat#173]
This was previously spotted in #70, as the example had such a struct, but the fix to that issue was merely to change the example so it did not rely on behaviour not yet implemented.
```
[cargo-make] INFO - Running Task: diplomat-gen-js
thread 'main' panicked at 'not yet implemented: Receiving structs that don't need a buffer', tool\src\js\conversions.rs:121:25
```

This is a problem for ICU4X, as the removal of the `sign_display` option from `ICU4XFixedDecimalFormatOptions` (as part of https://github.com/unicode-org/icu4x/issues/1267, see https://github.com/unicode-org/icu4x/pull/2025#discussion_r896016450) turns it into a single-field struct:
```diff
             ICU4XFixedDecimalFormatOptions {
                 grouping_strategy: ICU4XFixedDecimalGroupingStrategy::Auto,
-                sign_display: ICU4XFixedDecimalSignDisplay::Auto,
             }
```

**Comment by Manishearth** [comment]
Yeah we'll basically have to special case codegen there, because WASM treats these differently.

**Comment by Manishearth** [comment]
https://github.com/rust-diplomat/diplomat/blob/ce93a1b315f216a4369e2597c2b74006fe029b24/tool/src/js/conversions.rs#L124-L126

**Comment by ambiguousname** [comment]
Fixed by #742
